docs/source/virtual_ecosystem/implementation/animal_implementation.md

```diff
@@ -1,15 +1,196 @@
 ---
-jupyter:
-  jupytext:
-    cell_metadata_filter: all,-trusted
-    main_language: python
-    notebook_metadata_filter: settings,mystnb,language_info
-    text_representation:
-      extension: .md
-      format_name: markdown
-      format_version: '1.3'
-      jupytext_version: 1.16.4
+jupytext:
+  formats: md:myst
+  main_language: python
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.4
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+language_info:
+  codemirror_mode:
+    name: ipython
+    version: 3
+  file_extension: .py
+  mimetype: text/x-python
+  name: python
+  nbconvert_exporter: python
+  pygments_lexer: ipython3
+  version: 3.11.9
 ---
 
+# Animal Model Implementation
 
-# The Animal Model implementation
+## Model Overview
+
+The animal model in the Virtual Ecosystem platform represents the dynamics of animal
+communities. The implementation is structured
+around three core classes, each playing a distinct role in managing animal agents, their
+traits, and their interactions with the ecosystem. Together, these classes facilitate a
+flexible and scalable approach to simulating complex animal processes across spatial
+grids.
+
+### Core Classes
+
+#### **1. FunctionalGroup**
+
+The `FunctionalGroup` class encapsulates the fixed traits of an organism, defining its
+ecological role and life-history strategy. These traits include metabolic type, diet,
+taxonomic classification, reproductive strategy, and developmental type. Functional
+groups serve as templates for constructing animal cohorts and ensuring consistent trait
+inheritance across simulations.
+
+Key responsibilities:
+
+- Encodes fixed traits shared by cohorts within the group.
+- Provides scaling parameters for processes like territory size and prey selection.
```

Comment:
Are these all mass specific scaling factors? Can we unpack a bit: "Provides scaling factors that scale processes like territory size and prey selection to the mass of individuals within a cohort."